### ISSUE TITLE
Can't build jpeg-turbo-3.0.0

### ISSUE THREAD

**Issue by stalkerg** [rui314/mold#1117]
I can see the next error:
```
mold: fatal: simd/CMakeFiles/simd.dir/i386/jfdctflt-3dn.asm.o: RELA-type relocation table is not supported for this target
collect2: error: ld returned 1 exit status
```
mold version is 2.2.0

**Comment by rui314** [comment]
Are you building it for 32-bit x86 and not for 64-bit x86?

**Comment by stalkerg** [comment]
Yes, in Gentoo, we usually build such packages for 32 and 64-bit as well.

**Comment by rui314** [comment]
Did you know how `jfdctflt-3dn.asm.o` was built? It looks like it's a nasm assembly file, but I can't assemble that file myself.

**Comment by stalkerg** [comment]
Like this:
```
/usr/bin/nasm  -I/var/tmp/portage/media-libs/libjpeg-turbo-3.0.0/work/libjpeg-turbo-3.0.0_build-abi_x86_32.x86 -I/var/tmp/portage/media-libs/libjpeg-turbo-3.0.0/work/libjpeg-turbo-3.0.0 -DELF -DPIC -I"/var/tmp/portage/media-libs/libjpeg-turbo-3.0.0/work/libjpeg-turbo-3.0.0/simd/nasm/" -I"/var/tmp/portage/media-libs/libjpeg-turbo-3.0.0/work/libjpeg-turbo-3.0.0/simd/i386/" -g -MD simd/CMakeFiles/simd.dir/i386/jfdctflt-3dn.asm.o.d -MT simd/CMakeFiles/simd.dir/i386/jfdctflt-3dn.asm.o -f elf -o simd/CMakeFiles/simd.dir/i386/jfdctflt-3dn.asm.o /var/tmp/portage/media-libs/libjpeg-turbo-3.0.0/work/libjpeg-turbo-3.0.0/simd/i386/jfdctflt-3dn.asm
```

**Comment by rui314** [comment]
Thanks! So it looks like nasm creates RELA-type relocation tables for debug info sections, even though i386 psABI says that their relocation type is REL and not RELA. I think this is a nasm's bug. Can you report it to nasm developers? In the meantime, if you remove `-g` from the command line, nasm wouldn't create such RELA-type sections. (The relocations for the text section is correctly built as a REL relocation.)

```
Section Headers:
  [Nr] Name              Type            Addr     Off    Size   ES Flg Lk Inf Al
  [ 0]                   NULL            00000000 000000 000000 00      0   0  0
  [ 1] .data             PROGBITS        00000000 0002f0 000020 00  WA  0   0 32
  [ 2] .text             PROGBITS        00000000 000310 0002c0 00  AX  0   0 32
  [ 3] .debug_aranges    PROGBITS        00000000 0005d0 000020 00      0   0  1
  [ 4] .rela.debug_aranges RELA            00000000 0005f0 000018 0c     14   3  4
  [ 5] .debug_pubnames   PROGBITS        00000000 000610 000000 00      0   0  1
  [ 6] .debug_info       PROGBITS        00000000 000610 00004c 00      0   0  1
  [ 7] .rela.debug_info  RELA            00000000 000660 00003c 0c     14   6  4
  [ 8] .debug_abbrev     PROGBITS        00000000 0006a0 00001b 00      0   0  1
  [ 9] .debug_line       PROGBITS        00000000 0006c0 000101 00      0   0  1
  [10] .rela.debug_line  RELA            00000000 0007d0 00000c 0c     14   9  4
  [11] .debug_frame      PROGBITS        00000000 0007e0 000000 00      0   0  8
  [12] .debug_loc        PROGBITS        00000000 0007e0 000000 00      0   0  1
  [13] .shstrtab         STRTAB          00000000 0007e0 0000c4 00      0   0  1
  [14] .symtab           SYMTAB          00000000 0008b0 000110 10     15  15  4
  [15] .strtab           STRTAB          00000000 0009c0 0000c6 00      0   0  1
  [16] .rel.text         REL             00000000 000a90 000050 08     14   2  4
```

**Comment by stalkerg** [comment]
Interesting, I am not building such lib with debug symbols... maybe it's libjpeg bug as well.

**Comment by stalkerg** [comment]
Okey, it seems like it is the case only for nasm 2.15 and fixed in nasm 2.16 .
We can suppose it's just nasm bug. Close. Thanks!

**Comment by rui314** [comment]
Thank you for investigating! If you find any other issues, feel free to report them to us.
